[next] nilrt-snac: add integration ptests

amstewart opened the pull request:
### Summary of Changes

The nilrt-snac project now has an integration test suite.
* nilrt-snac. Add a run-ptest entrypoint and pytest subpackage to run it.
* packagegroup-ni-ptest-smoke. Add the `nilrt-snac-ptest` package.


### Justification

[AB#2776057](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2776057)


### Testing

TODO: Detail what testing has been done to ensure this submission meets requirements.

* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)
* [x] Tested the packagegroup on a cRIO VM.

```
admin@NI-cRIO-903x-VM-6ebe3712:~# ptest-runner -l
Available ptests:
nilrt-snac	/usr/lib/nilrt-snac/ptest/run-ptest
admin@NI-cRIO-903x-VM-6ebe3712:~# ptest-runner nilrt-snac
START: ptest-runner
2024-09-13T15:27
BEGIN: /usr/lib/nilrt-snac/ptest
============================= test session starts ==============================
platform linux -- Python 3.12.4, pytest-8.0.2, pluggy-1.4.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /usr/lib/nilrt-snac/tests/integration
configfile: pytest.ini
collecting ... collected 9 items

../tests/integration/test_cli.py::test_help PASSED                       [ 11%]
../tests/integration/test_cli.py::test_noargs PASSED                     [ 22%]
../tests/integration/test_cli.py::test_verify PASSED                     [ 33%]
../tests/integration/test_cli.py::test_version PASSED                    [ 44%]
../tests/integration/test_installation.py::test_conflicts_ipk PASSED     [ 55%]
../tests/integration/test_installation.py::test_iptables PASSED          [ 66%]
../tests/integration/test_installation.py::test_opkg_binary PASSED       [ 77%]
../tests/integration/test_installation.py::test_supported_distro PASSED  [ 88%]
../tests/integration/test_installation.py::test_wireguard_conf PASSED    [100%]

============================== 9 passed in 0.60s ===============================
DURATION: 1
END: /usr/lib/nilrt-snac/ptest
2024-09-13T15:27
STOP: ptest-runner
TOTAL: 1 FAIL: 0
```


### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

amstewart committed:
nilrt-snac: add integration ptests

The nilrt-snac project now has an integration test suite. Add a
run-ptest entrypoint and pytest subpackage to run it.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pkggrp-ni-smoke: use tabs for indentation

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
pkggrp-ni-ptest-smoke: add nilrt-snac-ptest

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

chaitu236 commented on recipes-ni/nilrt-snac/nilrt-snac_git.bb:
```diff
@@ -10,23 +10,38 @@ LIC_FILES_CHKSUM = "file://LICENSE;md5=380df876633ca23587b9851600778cc0"
 "
+
+FILES:${PN}-ptest += "${libdir}/${PN}/tests/integration"
+RDEPENDS:${PN}-ptest = "\
```
[ptest.bbclass ](https://github.com/ni/openembedded-core/blob/nilrt/master/next/meta/classes-recipe/ptest.bbclass#L26C1-L26C21)already adds RDEPENDS on ${PN}. Probably better to use a `+=` here and remove `nilrt-snac`.

amstewart replied:
Will fix.

chaitu236 replied:
Any reason to introduce a new commit to fix this rather than update the original commit 0b61b17445a45cd704f2dc1820423159f544b282?

amstewart replied:
Because there was nothing broken about the old commit - it was just suboptimal. Making a new commit is slightly more clear in the PR than force pushing new commit hashes.

chaitu236 commented on recipes-core/packagegroups/packagegroup-ni-ptest-smoke.bb:
```diff
@@ -31,6 +31,7 @@ RDEPENDS:${PN}:append = "\
 	nettle-ptest \
 	ni-hw-scripts-ptest \
 	ni-test-boot-time-ptest \
+	nilrt-snac-ptest \
```
Would these tests fail when snac configuration is not enabled or do they only check that files required to enable it are installed on the system? If the former, I don't think the ptest should be added here as we have test events that install this packagegroup and run `ptest-runner` which would run all these tests.

amstewart replied:
The integration tests do not change the system state and pass regardless of whether or not the system is in SNAC configuration. They are just verifying the correctness of the package installation and that the non-destructive high-level workflows (like --help, --version, and verify) function.

amstewart committed:
nilrt_snac: remove unneeded RDEPENDS

The ptest.bbclass already adds an RDEPENDS on `nilrt-snac`, but it is
overriden by the nilrt-snac-ptest RDEPENDS bb assignment.

Instead, append the ptest RDEPENDS.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V2
* Changed pytest RDEPENDS to `+=` and dropped the extraneous `nilrt-snac` RDEPENDS.
  * Rebuilt the recipe and verified that the package dependencies include `nilrt-snac`.
